src/providers/BackgroundBlurProvider/index.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  BackgroundBlurOptions,
+  BackgroundBlurVideoFrameProcessor,
+  BackgroundFilterSpec,
+  ConsoleLogger,
+  DefaultVideoTransformDevice,
+  Device,
+  LogLevel,
+  NoOpVideoFrameProcessor,
+  VideoFrameProcessor,
+} from 'amazon-chime-sdk-js';
+import React, {
+  createContext,
+  FC,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+interface Props {
+  /** The spec defines the assets that will be used for adding background blur to a frame */
+  spec?: BackgroundFilterSpec;
+  /** A set of options that can be supplied when creating a background blur video frame processor. */
+  options?: BackgroundBlurOptions;
+}
+
+interface BackgroundBlurProviderState {
+  createBackgroundBlurDevice: (
+    device: Device
+  ) => Promise<DefaultVideoTransformDevice>;
+  isBackgroundBlurSupported: boolean | undefined;
+}
+
+const BackgroundBlurProviderContext =
+  createContext<BackgroundBlurProviderState | null>(null);
```

Comment:
Can use `undefined` here as well I think, since initially it is not available and later becomes available.